pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,459 @@
+    # zenith -> solar_zenith
+    ParameterMap("zenith", "solar_zenith"),
+    # clearsky
+    ParameterMap("clearsky_dhi", "dhi_clear"),
+    ParameterMap("clearsky_dni", "dni_clear"),
+    ParameterMap("clearsky_ghi", "ghi_clear"),
+    ParameterMap("clearsky_gti", "poa_global_clear")
+]
+
+
+def get_solcast_tmy(
+    latitude, longitude, api_key, map_variables=True, **kwargs
+):
+    """Get irradiance and weather for a
+    Typical Meteorological Year (TMY) at a requested location.
+
+    Data is derived from a multi-year time series selected to present the
+    unique weather phenomena with annual averages that are consistent with
+    long term averages. See [1]_ for details on the calculation.
+
+    Parameters
+    ----------
+    latitude : float
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float
+        in decimal degrees, between -180 and 180, east is positive
+    api_key : str
+        To access Solcast data you will need an API key [2]_.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    kwargs:
+        Optional parameters passed to the API.
+        See [3]_ for full list of parameters.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        containing the values for the parameters requested.The times
+        in the DataFrame index indicate the midpoint of each interval.
+    metadata: dict
+        latitude and longitude of the request.
+
+    Examples
+    --------
+    >>> df, meta = pvlib.iotools.solcast.get_solcast_tmy(
+    >>>      latitude=-33.856784,
+    >>>      longitude=151.215297,
+    >>>      api_key="your-key"
+    >>> )
+
+    you can pass any of the parameters listed in the API docs,
+    like ``time_zone``. Here we set the value of 10 for
+    "10 hours ahead of UTC":
+
+    >>> df, meta = pvlib.iotools.solcast.get_solcast_tmy(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     time_zone=10,
+    >>>     api_key="your-key"
+    >>> )
+
+    References
+    ----------
+    .. [1] `Solcast TMY Docs <https://solcast.com/tmy>`_
+    .. [2] `Get an API Key <https://toolkit.solcast.com.au/register>`_
+    .. [3] `Solcast API Docs <https://docs.solcast.com.au/>`_
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        format="json",
+        **kwargs
+    )
+
+    data = _get_solcast(
+        endpoint="tmy/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+    return data, {"latitude": latitude, "longitude": longitude}
+
+
+def get_solcast_historic(
+    latitude,
+    longitude,
+    start,
+    api_key,
+    end=None,
+    duration=None,
+    map_variables=True,
+    **kwargs
+):
+    """Get historical irradiance and weather estimated actuals

```

Comment:
```suggestion
    """Get historical irradiance and weather estimated actuals.
```
I'm unfamiliar with the term "actuals" but maybe that is standard? As a non-native, I can't really tell what meaning is trying to be conveyed

Reply:
we refer to "live" and "historical" data as _estimated actuals_. These would be sometimes be called "measurements" I guess, but we avoid that term as we are really just estimating these values from satellite observations and not direct measurements of weather/power. Safe to assume that anyone using the Solcast API understands this?

Reply:
Hmm I think that's a very Solcast specific terminology. To be more in line with the other iotools I suggest removing mentions of "actuals" and just calling them "estimates"